### ISSUE TITLE
false positive no-unused-modules with Flow

### ISSUE THREAD

**Issue by Hypnosphi** [benmosher/eslint-plugin-import#1539]
When type import goes before default import from the same file, the default export is marked as unused:

```js
// foo.js

// @flow strict
export type Foo = string
export default ''
```

```js
// index.js

// @flow strict
import type {Foo} from './foo'
import foo from './foo'
```

> 3:1  error  exported declaration 'default' not used within other modules  import/no-unused-modules

It works OK if I change the imports order:
```js
// index.js

// @flow strict
import foo from './foo'
import type {Foo} from './foo'
```

**Comment by rfermann** [comment]
The issue is rather importing from the same source 2 times than importing a flow type. I already have a workaround in place to fix the current behaviour, will make a PR after #1542 is merged.

**Comment by Vages** [comment]
Now that #1542 is merged, do you have any plans to publish your workaround, @rfermann?
